Account/FirefoxAccount.swift

```diff
@@ -349,6 +355,19 @@ open class FirefoxAccount {
         return client.destroyDevice(ownDeviceId: ownDeviceId, withSessionToken: session.sessionToken as NSData) >>> succeed
     }
 
+    open func registerOrUpdateDevice(force: Bool = false) -> Success {
```

Comment:
Have this method take a `TokenState` as input. Down below, where you extracted this method, use `if let` to decide whether to call this, only binding `stateCache.value` once.